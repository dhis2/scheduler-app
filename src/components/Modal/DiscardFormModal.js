import React from 'react'
import PropTypes from 'prop-types'
import {
    Button,
    Modal,
    ModalContent,
    ModalActions,
    ButtonStrip,
} from '@dhis2/ui'
import i18n from '@dhis2/d2-i18n'
import history from '../../services/history'

const DiscardFormModal = ({ hideModal }) => (
    <Modal open small onClose={hideModal}>
        <ModalContent>
            {i18n.t('Are you sure you want to discard this form?')}
        </ModalContent>
        <ModalActions>
            <ButtonStrip end>
                <Button
                    name="cancel-discard-form"
                    secondary
                    onClick={hideModal}
                >
                    {i18n.t('Cancel')}
                </Button>
                <Button
                    name="discard-form"
                    destructive
                    onClick={() => {
                        hideModal()
                        history.push('/')
                    }}
                >
                    {i18n.t('Discard')}
                </Button>
            </ButtonStrip>
        </ModalActions>
    </Modal>
)

const { func } = PropTypes

DiscardFormModal.propTypes = {
    hideModal: func.isRequired,
}

export default DiscardFormModal
