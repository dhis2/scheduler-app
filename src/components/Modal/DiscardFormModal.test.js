import React from 'react'
import { shallow, mount } from 'enzyme'
import history from '../../services/history'
import DiscardFormModal from './DiscardFormModal'

jest.mock('../../services/history', () => ({
    push: jest.fn(),
}))

describe('<DiscardFormModal>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<DiscardFormModal hideModal={() => {}} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('calls hideModal when cancel button is clicked', () => {
        const spy = jest.fn()
        const wrapper = mount(<DiscardFormModal hideModal={spy} />)

        wrapper.find('button[name="cancel-discard-form"]').simulate('click')

        expect(spy).toHaveBeenCalled()
    })

    it('calls history push and hideModal when discard button is clicked', () => {
        const spy = jest.fn()
        const wrapper = mount(<DiscardFormModal hideModal={spy} />)

        wrapper.find('button[name="discard-form"]').simulate('click')

        expect(history.push).toHaveBeenCalledWith('/')
        expect(spy).toHaveBeenCalled()
    })

    it('calls hideModal when cover is clicked', () => {
        const spy = jest.fn()
        const wrapper = mount(<DiscardFormModal hideModal={spy} />)

        wrapper.find('.backdrop').simulate('click')

        expect(spy).toHaveBeenCalled()
    })
})
