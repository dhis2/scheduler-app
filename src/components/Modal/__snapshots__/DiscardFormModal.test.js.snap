// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscardFormModal> renders correctly 1`] = `
<Modal
  dataTest="dhis2-uicore-modal"
  onClose={[Function]}
  open={true}
  position="top"
  small={true}
>
  <ModalContent
    dataTest="dhis2-uicore-modalcontent"
  >
    Are you sure you want to discard this form?
  </ModalContent>
  <ModalActions
    dataTest="dhis2-uicore-modalactions"
  >
    <ButtonStrip
      dataTest="dhis2-uicore-buttonstrip"
      end={true}
    >
      <Button
        dataTest="dhis2-uicore-button"
        name="cancel-discard-form"
        onClick={[Function]}
        secondary={true}
        type="button"
      >
        Cancel
      </Button>
      <Button
        dataTest="dhis2-uicore-button"
        destructive={true}
        name="discard-form"
        onClick={[Function]}
        type="button"
      >
        Discard
      </Button>
    </ButtonStrip>
  </ModalActions>
</Modal>
`;
